packages/mask/src/components/shared/ApplicationBoard.tsx

```diff
@@ -178,14 +179,19 @@ function ApplicationBoardContent(props: Props) {
 
             {listedAppList.length > 0 ? (
                 <section
+                    ref={popperBoundaryRef}
                     className={cx(
                         classes.applicationWrapper,
                         recommendFeatureAppList.length > 2 && isCarouselReady() && isHoveringCarousel
                             ? classes.applicationWrapperWithCarousel
                             : '',
                     )}>
                     {listedAppList.map((application) => (
-                        <RenderEntryComponent key={application.entry.ApplicationEntryID} application={application} />
+                        <RenderEntryComponent
+                            key={application.entry.ApplicationEntryID}
+                            application={application}
+                            popperBoundary={popperBoundaryRef.current}
```

Comment:
Note: when the ref is initially null, then transition to a DOM, it WON'T update the component because it is declared by `useRef`.